scio-google-cloud-platform/src/test/scala/com/spotify/scio/bigquery/TableRowSyntaxTest.scala

```diff
@@ -17,21 +17,379 @@
 
 package com.spotify.scio.bigquery
 
-import org.scalatest.matchers.should.Matchers
+import com.google.common.io.BaseEncoding
+import com.spotify.scio.bigquery.types.{Geography, Json}
+import com.spotify.scio.coders.{Coder, CoderMaterializer}
+import org.apache.beam.sdk.util.CoderUtils
+import org.joda.time.{Instant, LocalDate, LocalDateTime, LocalTime}
+import org.scalactic.Equality
 import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import java.nio.charset.StandardCharsets
+import scala.jdk.CollectionConverters._
 
 class TableRowSyntaxTest extends AnyFlatSpec with Matchers {
-  "TableRowOps" should "#2673: not throw on get record" in {
-    val json = """{"record":{"foo":"bar"}}"""
-    val row = com.spotify.scio.util.ScioUtil.jsonFactory.fromString(json, classOf[TableRow])
-    val expected = Map("foo" -> "bar")
-    row.getRecord("record") shouldBe expected
-  }
-
-  it should "#3378: not throw an NPE on a non-existent subrecord" in {
```

Comment:
This is not very logical.
Added a `getRecordOpt` to be able to get `Option[TableRow]` instead